plugins/org.python.pydev/tests_completions/org/python/pydev/ast/codecompletion/PythonCompletionWithoutBuiltinsGrammar3Test.java

```diff
@@ -287,4 +287,112 @@ public void testTypingCast() throws Exception {
+    }
+
+    public void testUnion2() throws Exception {
+        String s;
+        s = "" +
+                "from typing import Union\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1(param:Union[A, B]):\n"
+                + "    pass\n"
+                + "def method2(param:Union[A, B]):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion3() throws Exception {
+        String s;
+        s = "" +
+                "from typing import Union as Foo\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1(param:Foo[A, B]):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion4() throws Exception {
+        String s;
+        s = "" +
+                "import typing\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1(param:typing.Union[A, B]):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion5() throws Exception {
+        String s;
+        s = "" +
+                "import typing as foo\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1(param:foo.Union[A, B]):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion6() throws Exception {
+        String s;
+        s = "" +
+                "import typing as foo\n"
+                + "class A(object):\n"
+                + "    def method_a(self):\n"
+                + "        pass\n"
+                + "class B(object):\n"
+                + "    def method_b(self):\n"
+                + "        pass\n"
+                + "def method1(param: foo.Union[A, B]):\n"
+                + "    param.me";
+        requestCompl(s, s.length(), -1, new String[] { "method_a()", "method_b()" });
+    }
+
+    public void testUnion7() throws Exception {
```

Comment:
One thing missing here is handling of unions with the new syntax, where the parameter is passed as:

```
def method2(param:A|B)
```

-- which is the same as a `Union[A, B]`.

Reply:
Supported in 11a8aab5538f4ba9232c1dd4334211812907a5c5 ;)